engine/apps/alerts/models/alert_group.py

```diff
@@ -612,6 +613,115 @@ def _update_metrics(self, organization_id, previous_state, state):
         """Update metrics cache for response time and state as needed."""
         update_metrics_for_alert_group.apply_async((self.id, organization_id, previous_state, state))
 
+    def update_state_by_backsync(
+        self, alert_receive_channel: "AlertReceiveChannel", new_state: AlertGroupState
+    ) -> None:
+        if self.state == new_state:
+            return
+
+        from apps.alerts.models import AlertGroupLogRecord
+
+        @dataclass
+        class StateUpdateData:
+            method: typing.Callable
+            log_type: int
+            kwargs: dict | None = None
+
+        @dataclass
+        class StateData:
+            remove_state: StateUpdateData
+            set_state: StateUpdateData
+
+        state_update_mapping = {
+            AlertGroupState.ACKNOWLEDGED: StateData(
+                set_state=StateUpdateData(
+                    method=self.acknowledge,
```

Comment:
hm.. I'm probably missing something but to be sure, you don't use the "_by_user" methods because...? passing `user=None, action_source=backsync` seems should work? and it will handle the metrics update, start escalation/ack reminder, log records, signal and dependent alert groups? maybe requiring some tuning? (e.g. I see we sometimes get org from user instead of the channel)

Reply:
I think using method "_by_user" without user looks a bit hacky (e.g. we would need to check if there is a user in the "_by_user" method to set `acknowledged_by=AlertGroup.USER` or `AlertGroup.SOURCE`) 🤔 
In our case methods "_by_source" with some tuning (adding action_source and some additional checks, related to logs) logically look better, but not every state action has this method.